src/resolve/resolved-dns-transaction.h

```diff
@@ -1,12 +1,24 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once
 
+
 #include "list.h"
 #include "resolved-def.h"
 #include "resolved-dns-dnssec.h"
 #include "resolved-dns-server.h"
 #include "resolved-forward.h"
 
+#if ENABLE_DNS_OVER_HTTPS
+#include <curl/curl.h>
+#endif
+
+#if ENABLE_DNS_OVER_HTTPS
+#include "curl-util.h"
+#endif
+
+#include "sd-event.h"
+#include "in-addr-util.h"
```

Comment:
order of inclusion is still not right. public-facing sd-*.h headers should be included before internal headers, which curl-util.h is